lib/id3c/cli/redcap.py

```diff
@@ -576,6 +576,42 @@ def id(self) -> str:
             raise RuntimeError(f"Primary record id field «{self.project.record_id_field}» not available on fetched record") from e
 
 
+    def update_record(self, record: Dict[str, str], date_format: str = "YMD", check_count: bool = True) -> int:
+        """
+        Updates one redcap record.
+
+        *record* must be a :py:class:``dict``s mapping REDCap field
+        names to values. The primary record id field and 
+        ``redcap_event_name`` are optional. Other pseudo-fields like
+        or ``redcap_repeat_instance`` may be necessary. All keys and
+         values must be strings.
+
+        Dates must be formatted as ``YYYY-MM-DD`` with the default
+        *date_format* of ``YMD``, as ``D/M/YYYY`` with ``DMY``, and as
+        ``M/D/YYYY`` with ``MDY``. Times are always assumed to be in the REDCap
+        server's timezone.
+
+        Any value provided for a field, including the empty string, will
+        overwrite any existing value.
+
+        This method is not suitable for creating a new record in projects that
+        use auto-numbered record ids.
+
+        Returns a count of the number of records updated, as reported by
+        REDCap which should be 1.
+        """
+        
+        record['record_id'] = self.id
+
+        # Event_name must be in raw format when Record object otherwise will return 400 Bad Request
+        if self.event_name:
+            if " " in self.event_name:
+                self.event_name = self.project.record(self.id, raw=True)[0]['redcap_event_name']
+            record['redcap_event_name'] = self.event_name
```

Comment:
@cooperqmarshall I think what be simplest here is to just make `redcap_event_name` an optional argument of `update_record` function so that it can be passed in (in the "raw" format) when needed.

Reply:
(The example I found where checking for a space would not work is HCT project, where we have event name labels "Enrollment" and "Encounter" which correspond to `enrollment_arm_1` and `encounter_arm_1`)